src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -57,6 +62,48 @@
   @Requirement
   private IModelAccessFacade modelAccess;
 
+  @Requirement
+  private List<IPageTypeCategoryRole> pageTypeCategoryList;
+
+  private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCategoryMap = new ConcurrentHashMap<>();
+
+  private ConcurrentMap<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+    if (typeNameToCategoryMap.isEmpty()) {
+      synchronized (typeNameToCategoryMap) {
+        if (typeNameToCategoryMap.isEmpty()) {
+          for (IPageTypeCategoryRole typeCategory : pageTypeCategoryList) {
+            IPageTypeCategoryRole beforeRegCat = typeNameToCategoryMap.putIfAbsent(
+                typeCategory.getTypeName(), typeCategory);
```

Comment:
ConcurrentMap is the better choice to ensure memory visibility compared to volatile.  The map could change if a Category-implementation decides to change the handled Category name. The Category names are not defined to be static. I am even not completely if we should consider the deprecated names in the map. Yet, I agree that we could return an unmodifiable map from the getter method, because it is a cache which must be rebuilded so far. I added resetTypeNameToCatCache to allow any Category implementation to reset the cache if needed. Done.